localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_callback.py

```diff
@@ -140,6 +140,17 @@ def _from_error(self, env: Environment, ex: Exception) -> FailureEvent:
             return self._get_callback_outcome_failure_event(env=env, ex=ex)
         return super()._from_error(env=env, ex=ex)
 
+    def _eval_body(self, env: Environment) -> None:
+        # Generate a TaskToken uuid within the context object, if this task resources has a waitForTaskToken condition.
+        if self._is_condition() and self.resource.condition == ResourceCondition.WaitForTaskToken:
+            task_token = env.context_object_manager.update_task_token()
+            env.callback_pool_manager.add(task_token)
+
+        super()._eval_body(env=env)
+
+        # Ensure the TaskToken field is reset, as this is only available during waitForTaskToken task evaluations.
+        env.context_object_manager.context_object["Task"] = None
```

Comment:
Does setting it to `None` have the same effect as deleting it from the dict? (just wondering)

Reply:
It has the same affect, however I agree it would be better to have the field missing to remove ambiguity. I have changed the annotation from Optional to NotRequired, and pop this field at this depth. I think it's a lot clearer now, and would like to expand this pattern in other areas of the interpreter